docs/developer/feeds.rst

```diff
@@ -4,122 +4,199 @@
+so feed names must follow standard
+`URI formatting rules <https://crossbar.io/docs/URI-Format/>`_, meaning
+the feed can only use lowercase letters, numbers, and underscores.
+As the ``agent_uri`` will be unique for each agent, feed names are not required
+to be globally unique.
 
-        Args:
-            agent_address (string):
-                agent address of the feed.
+For instance, the Lakeshore372 agent may register a feed called ``temperatures``.
+A Lakeshore372 instance with instance-id ``LSASIM`` would then publish data to
+the URI ``observatory.LSASIM.feeds.temperatures``.
 
-            feed_name (string):
-                name of the feed.
-        """
+Agg Params
+```````````
+If you'd like your feed to be recorded by the hk aggregator, you must register
+with the keyword ``record=True``, and you can customize the hk aggregator
+behavior with the ``agg_params`` option.
+The following options can be specified here:
 
-        def _data_handler(_data):
-            """Callback whenever data is published to an aggregated feed"""
-            data, feed = _data
-            if self.aggregate:
-                self.incoming_data[feed["address"]].append(data)
+.. list-table:: Aggregator params
+    :widths: 20 20
 
-        feed_address = "{}.feeds.{}".format(agent_address, feed_name)
-        if feed_address in self.incoming_data.keys():
-            return
+    * - frame_length
+      - Aggregation time (seconds) before frame is written to disk
 
-        self.agent.subscribe_to_feed(agent_address, feed_name, _data_handler)
-        self.incoming_data[feed_address] = []
-        self.log.info("Subscribed to feed {}".format(feed_address))
+    * - fresh_time
+      - Time (seconds) before feed is considered "stale", and is removed from
+        the HK status frame
 
-This function is called whenever the aggregator needs to subscribe to a new feed.
-It subscribes to the agent's feed with the callback ``_data_handler``,
-which appends the received data to that feed's timestream.
+Publishing to a Feed
+----------------------
+You can publish data to a feed by calling::
 
-``subscribe_to_feed`` by default protects against an agent subscribing to the
-same feed multiple times. This is in case you want to run a client script
-multiple times without restarting the agent, which can call
-``subscribe_to_feed`` multiple times for the same feed. If you purposefully want
-to subscribe to a feed more than once, you can set ``force_subscribe=True``.
+    agent.publish_to_feed(feed_name, message)
 
+For standard feeds, message can be any json-ifyable object (i.e. strings, ints,
+floats, bools, or dicts containing these).
+Callbacks will receive the tuple ``(message, feed_data)`` where ``feed_data``
+is a dict encoding most OCS.Feed attributes.
+
+
+.. _recorded_feed_registration:
+
+Recorded Feed Registration
+````````````````````````````
+To make sure that a feed is picked up
+by the aggregator, it must be registered with the option 'record=True'.
```

Comment:
Is there any instance/example for registering a feed where you want record=False?

Reply:
Yep! Heartbeats are OCS feeds. Also pysmurf-monitor publishes data to a non-recorded feed so that the pysmurf-controller can pick it up and put it into session data.

Reply:
Yep! Heartbeats are OCS feeds. Also pysmurf-monitor publishes data to a non-recorded feed so that the pysmurf-controller can pick it up and put it into session data.